viewing_party/party.py

```diff
@@ -1,23 +1,165 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    movie = {}
+    movie_set = {title, genre, rating}
+
+    if None in movie_set:
+        return None
+    else:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+    
+    return movie
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+    for movie in user_data["watchlist"]:
+        if title != movie["title"]:
+            continue
+        else:
+            movie_index = user_data["watchlist"].index(movie)
+            removed_movie = user_data["watchlist"].pop(movie_index)
+            add_to_watched(user_data, removed_movie)
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    total = 0
+
+    if not user_data["watched"]:
+        return 0.0
+    
+    for movie in user_data["watched"]:
+        total += movie["rating"]
+    
+    average = total / len(user_data["watched"])
+
+    return average
+
+def get_most_watched_genre(user_data):
+    
+    compare_dict = {}
+
+    if not user_data["watched"]:
+        return None
+
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre not in compare_dict:
+            compare_dict[genre] = 0
+        compare_dict[genre] += 1
+    
+    most_watched_genre = max(compare_dict, key=compare_dict.get)
+
+    return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+
+    friends_movie_list = []
+
+    for friend in user_data["friends"]:
+        friends_movie_list.extend(friend["watched"])
+    
+    unique_movies = []
+
+    for movie in user_data["watched"]:
+        if movie not in friends_movie_list:
+            unique_movies.append(movie)
+
+    return unique_movies
+
+def get_friends_unique_watched(user_data):
+
+    friends_movie_list = []
+
+    for friend in user_data["friends"]:
+        friends_movie_list.extend(friend["watched"])
+    
+    unique_movies = []
+
+    for movie in friends_movie_list:
+        if movie not in user_data["watched"] and movie not in unique_movies:

```

Comment:
Here we have two separate `in` list checks. How might we think about improving this with set lookups? Since we do want to return a list, we can't simply replace `unique_movies` with a set, but we could make a parallel set so that each time we add a movie to the result, we add its title to the set. Then we could change the `not in unique_movies` check to use the unique title set instead.